src/ucp/core/ucp_context.c

```diff
@@ -1831,7 +1834,8 @@ static void ucp_apply_params(ucp_context_h context, const ucp_params_t *params,
         ucs_snprintf_zero(context->name, UCP_ENTITY_NAME_MAX, "%s",
                           params->name);
     } else {
-        ucs_snprintf_zero(context->name, UCP_ENTITY_NAME_MAX, "%p", context);
+        ucs_snprintf_zero(context->name, UCP_ENTITY_NAME_MAX, "ucp_context_%lu",
+                          ucs_atomic_fadd64(&context_counter, 1));
```

Comment:
why is atomic needed here?

Reply:
in case multiple threads call ucp_init at the same time. context_counter is a global variable